peartree/summarizer.py

```diff
@@ -245,16 +245,35 @@ def generate_edge_and_wait_values(
     ftrips = feed.trips.copy()
     ftrips = ftrips[~ftrips['route_id'].isnull()]
 
+    # Trim down stop_times df based on requested time range
+    # before feeding into the interpolation step downstream
+    # Make a copy
+    init_stop_times = feed.stop_times.copy()
+
+    # Create masks for time range
+    start_time_mask = (init_stop_times.arrival_time >= target_time_start)
```

Comment:
Was there a plan to add an hour or subtract an hour from each end to capture the full window of time and include stops that need to be interpolated?

Reply:
sorry. the comments are confusing. let me explain the idea here:

The general idea is to find trips with at least one stop time inside the requested time range (say 7:00 am to 9:00 am). That means, in the end, I am tossing all the trips that occurred COMPLETELY outside the time range. 

In other words, it is possible that among the trips that are preserved, there are some trips that started before 9:00 am, and stopped at the first two stops before 9:00 am, and the rest are after 9:00 am. Though theoretically, we do not need them, i think it is fine to keep them. Since we have tossed majority of the unneeded stop times with some rough trimming.

So what I added are:
1. Filter the full `stop_times` df with time range 
2. Find trip IDs in the filtered `stop_times` df, which are trips with at least one stop time inside the time range
3. Use the above trip ID list to filter the full `stop_times` df again.
4. Feed the above filtered one to the stop time interpolation step.

Reply:
I don't think adding or subtracting hour will help because it will still get rid of all the stop times that are `NaN`

Reply:
I see!
Thank you for the explanation. That makes total sense. So we consider that whole leg of the trip, so long as one end of it falls within the allowed timeframe. This makes total sense. Apologies for the confusion on my end.